packages/server/src/api/routes/tests/automation.spec.ts

```diff
@@ -121,6 +123,78 @@ describe("/automations", () => {
       expect(events.automation.stepCreated).toHaveBeenCalledTimes(2)
     })
 
+    it("Should ensure you can't have a branch as not a last step", async () => {
+      const automation = branchAutomationIncorrectPosition()
+      const res = await request
+        .post(`/api/automations`)
+        .set(config.defaultHeaders())
+        .send(automation)
+        .expect("Content-Type", /json/)
+        .expect(400)
+
+      expect(res.body.message).toContain("must contain at least 1 items")
+    })
+
+    it("Should check validation on an automation that has a branch step with no children", async () => {
+      const automation = branchAutomationIncorrectPosition()
+      automation.definition.steps[0].inputs.branches = [
+        { name: "test", condition: { equal: { "steps.1.success": "true" } } },
+      ]
+      automation.definition.steps[0].inputs.children = {}
+
+      const res = await request
+        .post(`/api/automations`)
+        .set(config.defaultHeaders())
+        .send(automation)
+        .expect("Content-Type", /json/)
+        .expect(400)
```

Comment:
I'd add this post call to `packages/server/src/tests/utilities/api/automation.ts` and use the `Expectations` type you'll find reference to in there to assert the shape of the body (e.g. `{ message: "Branch steps are only allowed as the last step" }`. This should make these tests shorter and easier to parse, also more in line with the bulk of our other integration tests.